app-creator/data/templates/templates.go

```diff
@@ -0,0 +1,50 @@
+package templates_data 
+
+import (
+	"encoding/json"
+	"io"
+	"context"
+	"cloud.google.com/go/datastore"
+	"log"
+)
+
+type Template struct {
+	Id string `json:"id"` 
+	Name string `json:"name"`
+	ImageUrl string `json:"imageUrl"`
+	Template string `json:"template"`
+}
+
+type Templates []*Template
+
+/**
+* Converts an array of Template data into JSON.
+*/
+func (t *Templates) ToJSON(w io.Writer) error {
+	encoder := json.NewEncoder(w)
+	return encoder.Encode(t)
+}
+
+/**
+* Converts a JSON representation of a Template object into a Template instance. 
+*/
+func (t *Template) FromJSON(r io.Reader) error {
+	e := json.NewDecoder(r)
+	return e.Decode(t)
+}
+
+/**
+* Returns array of templates.
+*/
+func GetTemplates(db *datastore.Client, l *log.Logger, ctx context.Context) Templates {
+	templates := []*Template{}
+	_, err := db.GetAll(ctx, datastore.NewQuery("Template"), &templates)
```

Comment:
We may want to support a one-time operation to load up the datastore with the default templates if none exist, since users who build the app creator locally will have fresh datastores.

Reply:
Since we're are making changes to the template structure, I will add this in a following PR.

Reply:
issue tracked here https://github.com/googleinterns/earthengine-community/projects/1#card-41793493